projects/valtimo/process-link/src/lib/components/select-url/select-url.component.ts

```diff
@@ -0,0 +1,194 @@
+                } else {
+                    this.buttonService.disableSaveButton();
+                }
+            })
+
+        this.stateService.url$
+            .subscribe(url => this.url.setValue(url));
+
+        this.processLinkService.getVariables()
+            .subscribe(variables => this._variables$.next(variables.variables))
+    }
+
+    ngOnDestroy(): void {
+        this._subscriptions.unsubscribe();
+    }
+
+    private openBackButtonSubscription(): void {
+        this._subscriptions.add(
+            this.buttonService.backButtonClick$.subscribe(() => {
+                this.stateService.setInitial();
+            })
+        );
+    }
+
+    private openSaveButtonSubscription(): void {
+        this._subscriptions.add(
+            this.buttonService.saveButtonClick$.subscribe(() => {
+                this.stateService.startSaving();
+                this.saveProcessLink();
+            })
+        );
+    }
+
+    private saveProcessLink(): void {
+        this.stateService.selectedProcessLink$.pipe(take(1)).subscribe(selectedProcessLink => {
+            if (selectedProcessLink) {
+                this.updateProcessLink();
+            } else {
+                this.saveNewProcessLink();
+            }
+        });
+    }
+
+    private updateProcessLink(): void {
+        this.stateService.selectedProcessLink$
+            .pipe(take(1))
+            .subscribe((selectedProcessLink) => {
+                const updateProcessLinkRequest: URLProcessLinkUpdateRequestDto = {
+                    id: selectedProcessLink.id,
+                    url: this.url.value,
+                };
+
+                this.processLinkService.updateProcessLink(updateProcessLinkRequest).subscribe(
+                    () => {
+                        this.stateService.closeModal();
+                    },
+                    () => {
+                        this.stateService.stopSaving();
+                    }
+                );
+            });
+    }
+
+    private saveNewProcessLink(): void {
+        combineLatest([
+            this.stateService.modalParams$,
+            this.stateService.selectedProcessLinkTypeId$
+        ])
+            .pipe(
+                take(1),
+                switchMap(([modalParams, processLinkTypeId]) =>
+                    this.processLinkService.saveProcessLink({
+                        url: this.url.value,
+                        activityType: modalParams.element.activityListenerType,
+                        processDefinitionId: modalParams.processDefinitionId,
+                        processLinkType: processLinkTypeId,
+                        activityId: modalParams.element.id
+                    })
+                )
+            )
+            .subscribe(
+                () => {
+                    this.stateService.closeModal();
+                },
+                () => {
+                    this.stateService.stopSaving();
+                }
```

Comment:
This sort of writing is deprecated, check the updated way to write the subscribe